File: News-Android-App/src/main/java/de/luhmer/owncloudnewsreader/helper/URLConnectionReader.java
package de.luhmer.owncloudnewsreader.helper;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.net.URL;
import java.net.URLConnection;

/**
 * Created by David on 13.01.2016.
 */
public class URLConnectionReader {
    public static String getText(String url) throws Exception {
        URL website = new URL(url);
        URLConnection connection = website.openConnection();
        BufferedReader in = new BufferedReader(
                new InputStreamReader(
                        connection.getInputStream()));

        StringBuilder response = new StringBuilder();
        String inputLine;

        while ((inputLine = in.readLine()) != null)
            response.append(inputLine);

        in.close();

        return response.toString();
    }
}

File: News-Android-App/src/main/java/de/luhmer/owncloudnewsreader/helper/URLConnectionReader.kt
@file:JvmName("URLConnectionReader")

package de.luhmer.owncloudnewsreader.helper

import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader
import java.net.URL

/**
 * Created by David on 13.01.2016.
 */
@Throws(IOException::class)
fun getText(url: String?): String {
    val website = URL(url)
    val connection = website.openConnection()

    val response = StringBuilder()
    BufferedReader(InputStreamReader(connection.getInputStream())).use { inReader ->
        {
            var inputLine: String?
            while (inReader.readLine().also { inputLine = it } != null) response.append(inputLine)
        }
    }
    return response.toString()
}
